ansible/roles/set-repositories/tasks/rhn-repos.yml

```diff
@@ -1,41 +1,31 @@
 # vim: set ft=ansible:
 ---
+
 - name: Force unregister before register
-  redhat_subscription:
-    state: absent
-  register: task_result
-  until: task_result | succeeded
-  retries: 10
-  delay: 5
+  command: "{{ item }}"
+  args:
+    warn: False
+  with_items:
+  - 'subscription-manager clean'
+  - 'subscription-manager remove --all'
+  - 'yum remove -y "katello-ca-consumer-*"'
 
-- name: register node with subscription-manager
+- name: 'Register system using Red Hat Subscription Manager'
   redhat_subscription:
     state: present
     username: "{{ rhel_subscription_user }}"
     password: "{{ rhel_subscription_pass }}"
-    autosubscribe: false
-  register: task_result
-  until: task_result | succeeded
-  retries: 10
-  delay: 5
+    pool_ids: "{{ rhsm_pool_ids }}"
+    auto_attach: false
 
-# TODO: should take pool ids from a var
-- name: get product pool id
-  shell: /usr/bin/subscription-manager list --all --available --matches="*{{rhn_pool_id_string}}*" | awk '/Pool ID/ {print $3}' | head -1
-  # command: subscription-manager list --all --available --matches="OpenShift Container Platform" | awk '/Pool ID/ {print $3}' | head -1
-  register: pool_id
-  until: pool_id | succeeded
-  retries: 10
-  delay: 5
-
-- name: attach host to subscription pool
-  shell: /usr/bin/subscription-manager attach --pool={{ pool_id.stdout }}
-  register: task_result
-  until: task_result.rc == 0
-  retries: 10
-  delay: 5
-
-- name: enable repos for rhel
-  command: subscription-manager repos --enable "{{ item }}"
+- name: "Build command line for repos to enable"
+  set_fact:
+    repos_params: "{{ repos_params|default('') }} --enable={{ item }}"
```

Comment:
You should not use the same variable `repos_params`.

`set_fact` won't append the `--enable={{ item }}` if the `repos_params` is set via extra-vars.
When set as extra-vars, it takes precedence over `set_fact`.

Reply:
`repos_params` is only used during run-time for this playbook, is not meant to be used in extra-vars while using the role, the repositories related variable provided by the user is `rhel_repos`. Does it make sense?

Reply:
OK!
I'll give a try and let you know!

Reply:
Great, as an example for the actual config I'm developing these are the RHSM related variables:

```repo_method: rhn
use_subscription_manager: true
rhel_subscription_user: "xxxx@redhat.com"
rhel_subscription_pass: "xxxxxxxx"
rhsm_pool_ids:
  - "xxxxxxxxxxxxxxxxxxxxxxxxxxxx"
rhel_repos:
   - "rhel-8-for-x86_64-baseos-rpms"
   - "rhel-8-for-x86_64-appstream-rpms"
   - "ansible-2-for-rhel-8-x86_64-rpms"```